tests/components/sensor/test_pilight.py

```diff
@@ -0,0 +1,120 @@
+"""The tests for the Pilight sensor platform."""
+import logging
+
+from homeassistant.bootstrap import setup_component
+import homeassistant.components.sensor as sensor
+from homeassistant.components import pilight
+
+from tests.common import get_test_home_assistant, assert_setup_component
+
+HASS = None
+
+
+def fire_pilight_message(protocol, data):
+    """Fire the fake pilight message."""
+    message = {pilight.ATTR_PROTOCOL: protocol}
+    message.update(data)
+    HASS.bus.fire(pilight.EVENT, message)
+
+
+def setUpModule():   # pylint: disable=invalid-name
+    """Initialize a Home Assistant server."""
+    global HASS
+
+    HASS = get_test_home_assistant()
+    HASS.config.components = ['pilight']
+
+
+def tearDownModule():   # pylint: disable=invalid-name
+    """Stop the Home Assistant server."""
+    HASS.stop()
+
+
+def test_sensor_value_from_code():
+    """Test the setting of value via pilight."""
+    with assert_setup_component(1):
```

Comment:
Once a component is in the `HASS.config.components` list it will never be setup again

Like you mentioned, your use of `setUpModule` only executes once and you only have a single HASS instance throughout

To get around this:
- Rename `setUpModule` to `setup_method(self, _)` or `setUp(self)`. This is quite common in most HASS unit tests
- Before `setup_component` remove the component from `HASS.config.components` - use with some care/knowledge of your setup procedure and that there is nothing else that is keeping some state